task/sast-shell-check/0.1/sast-shell-check.yaml

```diff
@@ -0,0 +1,179 @@
+      env:
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: PROJECT_NVR
+          value: $(params.PROJECT_NVR)
+        - name: RECORD_EXCLUDED
+          value: $(params.RECORD_EXCLUDED)
+        - name: IMP_FINDINGS_ONLY
+          value: $(params.IMP_FINDINGS_ONLY)
+      script: |
+        #!/usr/bin/env bash
+        set -x
+        . /utils.sh
+
+        # FIXME: once https://github.com/konflux-ci/konflux-test/pull/303 is available in latest image,
+        # work around below is no longer needed
+        microdnf install -y ShellCheck
+        microdnf download csmock-plugin-shellcheck && rpm -ivh --nodeps ./csmock-plugin-shellcheck-*.rpm && rm -f ./csmock-plugin-shellcheck-*.rpm
+
+        OUTPUT_FILE="shellcheck-results.json"
+        SOURCE_CODE_DIR=$(workspaces.workspace.path)/source
+
+        # generate all shellcheck result JSON files to $SC_RESULTS_DIR, which defaults to ./shellcheck-results/
+        /usr/share/csmock/scripts/run-shellcheck.sh "$SOURCE_CODE_DIR"
+
+        CSGREP_OPTS=(
+            --mode=json
+            --strip-path-prefix="$SOURCE_CODE_DIR"/
+            --event='error|warning'
+            --remove-duplicates
+            --embed-context=3
+        )
+        csgrep "${CSGREP_OPTS[@]}" ./shellcheck-results/*.json > "$OUTPUT_FILE"
+
+        if [ $? -ne 0 ]; then
+            echo "Error occurred while running 'run-shellcheck.sh'"
+            note="Task $(context.task.name) failed: For details, check Tekton task log."
+            ERROR_OUTPUT=$(make_result_json -r ERROR -t "$note")
+            echo "${ERROR_OUTPUT}" | tee "$(results.TEST_OUTPUT.path)"
+            exit 1
+        fi
+
+        # Filter known false positives if KFP_GIT_URL is set
+        if [ -n "${KFP_GIT_URL}" ]; then
+            echo "Filtering known false positives using ${KFP_GIT_URL}"
+
+            # build initial csfilter-kfp command
+            csfilter_cmd="csfilter-kfp --verbose --kfp-git-url=${KFP_GIT_URL}"
+
+            # Append --project-nvr option if PROJECT_NVR is set
+            if [ -n "${PROJECT_NVR}" ]; then
+                csfilter_cmd+=" --project-nvr=${PROJECT_NVR}"
+            fi
+
+            if [[ "${RECORD_EXCLUDED}" == "true" ]]; then
+                csfilter_cmd+=" --record-excluded=excluded-findings.json"
+            fi
+
+            # Execute the command and capture any errors
+            if ! "${csfilter_cmd[@]}" "$OUTPUT_FILE" > "${OUTPUT_FILE}.filtered" 2> "${OUTPUT_FILE}.error"; then
+                echo "Error occurred while filtering known false positives:"
+                cat "${OUTPUT_FILE}.error"
+                note="Task $(context.task.name) failed: For details, check Tekton task log."
+                ERROR_OUTPUT=$(make_result_json -r ERROR -t "$note")
+                echo "${ERROR_OUTPUT}" | tee "$(results.TEST_OUTPUT.path)"
+                rm -f "${OUTPUT_FILE}.filtered" "${OUTPUT_FILE}.error"
```

Comment:
I do not think you need to explicitly remove any files before exiting.  The whole overlay file system is going to be dropped when the container exits.

Reply:
removed